src/jit/lowerxarch.cpp

```diff
@@ -925,7 +925,8 @@ void Lowering::TreeNodeInfoInit(GenTree* tree)
     // the below logic leads to conservative register specification.  In future when or if we find
     // one such case, this logic needs to be fine tuned for that case(s).
     if (varTypeIsByte(tree) || ((tree->OperGet() == GT_CAST) && varTypeIsByte(tree->CastToType())) ||
-        (tree->OperIsCompare() && varTypeIsByte(tree->gtGetOp1()) && varTypeIsByte(tree->gtGetOp2())))
+        (tree->OperIsCompare() && varTypeIsByte(tree->gtGetOp1()) &&
+         (varTypeIsByte(tree->gtGetOp2()) || tree->gtGetOp2()->IsCnsIntOrI())))
```

Comment:
I have seen one such case in the past.  AFAIK, there is nothing in our implementation that ensures that op1 is never a constant.